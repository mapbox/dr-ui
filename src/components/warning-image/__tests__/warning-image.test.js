import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './warning-image-test-cases.js';

describe('warning-image', () => {
  describe(testCases.warning.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.warning;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.error.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.error;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.bigger.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.bigger;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.biggerRed.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.biggerRed;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
