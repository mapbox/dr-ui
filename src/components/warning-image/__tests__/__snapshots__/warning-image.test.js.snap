// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`warning-image Basic renders as expected 1`] = `
<div
  style={
    Object {
      "height": "60px",
      "padding": "5px",
      "width": "60px",
    }
  }
>
  <div
    className="bg-blue-light px3 round-full"
  >
    <svg
      className="color-blue"
      viewBox="0 0 18 18"
    >
      <title>
        warning
      </title>
      <path
        d="M9,3C8.4,3,7.8,3.3,7.5,3.8L3.2,13c-0.5,0.8,0,2,1.1,2H9h4.7c1.1,0,1.6-1.2,1.1-2l-4.3-9.2C10.2,3.3,9.6,3,9,3z M9,6
c0.6,0,1,0.4,1,1v2c0,0.6-0.4,1-1,1S8,9.6,8,9V7C8,6.4,8.4,6,9,6z M9,11c0.6,0,1,0.4,1,1s-0.4,1-1,1s-1-0.4-1-1S8.4,11,9,11z"
        style={
          Object {
            "fill": "currentColor",
          }
        }
      />
    </svg>
  </div>
</div>
`;
